vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go

```diff
@@ -284,11 +285,11 @@ func (feeder *clusterStateFeeder) GarbageCollectCheckpoints() {
 
 	for _, namespaceItem := range namespaceList.Items {
 		namespace := namespaceItem.Name
-		checkpointList, err := feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).List(context.TODO(), metav1.ListOptions{})
+		checkpointList, err := feeder.vpaCheckpointLister.VerticalPodAutoscalerCheckpoints(namespace).List(labels.Everything())
```

Comment:
I think now that we have a `vpaCheckpointLister` inside the `cluster_feeder`, we no longer need the list of namespaces here.
Instead, we can just list _all checkpoints_ (already filtered by `vpaObjectNamespace`, if that is set) above like this
```
checkpointList, err := feeder. vpaCheckpointLister.List(labels.Everything())
```

And then iterate over those checkpoints and filter them, if they are in a namespace that is in `vpaIgnoredNamespaces`. Alternatively, we could do the filtering once on the entire list, similar to what is done in `filterVPAs`